toolkit/scripts/test_package_install.sh

```diff
@@ -0,0 +1,199 @@
+    if [[ $ARCHITECTURE == "x86_64" ]]; then
+        DAILY_BUILD_ARCH="-x86-64"
+    else
+        DAILY_BUILD_ARCH="-aarch64"
+    fi
+    LKG_BASE_URL=https://mariner3dailydevrepo.blob.core.windows.net/daily-repo-$DAILY_BUILD_ID$DAILY_BUILD_ARCH
+fi
+
+echo "---------------------------------------------------------------"
+echo "-- Install packages in provided folder --"
+echo "---------------------------------------------------------------"
+echo "-- RPM_DIRECTORY                            -> $RPM_DIRECTORY"
+echo "-- RPMS_TO_INSTALL                          -> $RPMS_TO_INSTALL"
+echo "-- LKG_BASE_URL                             -> '$LKG_BASE_URL'"
+echo "-- DAILY_BUILD_ID                           -> '$DAILY_BUILD_ID'"
+echo "-- DAILY_BUILD_ARCH                         -> '$DAILY_BUILD_ARCH'"
+echo "-- DIST_TAG                                 -> '$DIST_TAG'"
+echo "-- PACKAGES_TO_NOT_INSTALL                  -> $PACKAGES_TO_NOT_INSTALL"
+echo "-- PACKAGES_TO_NOT_UNINSTALL                -> $PACKAGES_TO_NOT_UNINSTALL"
+echo ""
+
+echo "##[debug]Installing these space separated packages:"
+echo "$RPMS_TO_INSTALL"
+echo ""
+
+RPMS_TO_INSTALL=($RPMS_TO_INSTALL)
+PACKAGES_TO_NOT_INSTALL=($PACKAGES_TO_NOT_INSTALL)
+PACKAGES_TO_NOT_UNINSTALL=($PACKAGES_TO_NOT_UNINSTALL)
+
+
+exists_in_list() {
+    local VALUE=$1
+    shift
+    local LIST=("$@")
+
+    for x in "${LIST[@]}"; do
+        if [ "$x" = "$VALUE" ]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# Add built packages as a local repo for tdnf
+sudo createrepo --compatibility "$RPM_DIRECTORY"
+
+repo_content=$(cat << EOF
+[builtpackages]
+name=Local Built Packages($ARCHITECTURE)
+baseurl=file://$RPM_DIRECTORY
+enabled=1
+priority=0
+skip_if_unavailable=True
+enabled=1
+EOF
+)
+
+# create the repo metadata file
+echo "$repo_content" | sudo tee /etc/yum.repos.d/builtpackages.repo > /dev/null
+
+echo "##[debug]Created repo file with content:"
+sudo cat /etc/yum.repos.d/builtpackages.repo
+echo ""
+
+# Add daily 3.0 repo if available
+repo_content_daily=$(cat << EOF
+[daily3]
+name=Daily 3.0 repo
+baseurl=$LKG_BASE_URL
+gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY
+gpgcheck=0
+repo_gpgcheck=0
+enabled=1
+skip_if_unavailable=True
+sslverify=0
+EOF
+)
+
+if [[ $DIST_TAG == ".azl3" ]]; then
+    echo "$repo_content_daily" | sudo tee /etc/yum.repos.d/daily3.repo > /dev/null
+    echo "##[debug]Created daily repo file with content:"
+    sudo cat /etc/yum.repos.d/daily3.repo
+    echo ""
+else
+    echo "Mariner 3.0 was not detected, so skipping daily repo setup"
+fi
+
+# update the cached binary metadata.
+sudo tdnf makecache
+
+sudo tdnf repolist --refresh
+
+PACKAGES_FAIL_INSTALL=()
+PACKAGES_FAIL_UNINSTALL=()
+
+for rpm_to_install in "${RPMS_TO_INSTALL[@]}"; do
+    path_to_rpm=$(find $RPM_DIRECTORY -name "$rpm_to_install")
+    package_name_to_install=$(rpm -qp --queryformat "%{name}" "$path_to_rpm" 2>/dev/null)
```

Comment:
Skipping the version-release_no combined with `skip_if_unavailable=True` may bring the package from remote! It's a false positive.